toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/MainApp.java

```diff
@@ -11,246 +11,236 @@
+		execute();
 	}
 	
-	public WebConfig getWebConfig() {
-		return webConfigProvider.get();
+	private void execute() {
+		try {
+			if(hasPluginDirectory() && hasScriptsDirectory()) {
+				NotificationManager.showMessage(MainAppMessages.SCRIPTS_EXECUTION).showNotification();
+				Thread thread = new Thread(new WaiterThread(webConfigProvider));
+				thread.start();
+			}
+		} catch (IOException e1) {
+			LOG.error(e1.getMessage(), e1);
+		}
+	}
+
+	private boolean hasPluginDirectory() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_PLUGIN_DIR);
 	}
 
-	public boolean verificationWebApp(String nomUrlATester) throws IOException{
-		String input = "";
-		if(nomUrlATester == webAppName) {
-			input = webConfigProvider.get().getWebAppUrl();
+	private boolean hasScriptsDirectory() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_SCRIPTS_DIR);
+	}
+
+	private void killListener(ActionEvent event){
+		try {
+			agentServer.unRegister();
+		} catch (UnknownHostException exception) {
+			LOG.error(exception.getMessage(), exception);
+		}finally {
+			System.exit(0);
 		}
-		if(nomUrlATester == recorderName) {
-			input = webConfigProvider.get().getWebInitRecordingUrl();
+	}
+
+	private void start(ActionEvent e) {
+		try {
+    		if(connectedToWebApp) {
+				if(hasValidRecordingEnvironment()) {
+					listenerStarted = true;
+					browserManager.startRecording();
+				}
+				else {
+					NotificationManager.showMessage(MainAppMessages.UNABLE_START_RECORDER).showNotification();
+				}
+    		}
+    		else {
+				NotificationManager.showMessage(MainAppMessages.UNABLE_CONNECT_WEBAPP).showNotification();
+    		}
+		} catch (IOException e1) {
+			LOG.error(e1.getMessage(), e1);
 		}
-		if(nomUrlATester == chromeDriverName) {
-			input = webConfigProvider.get().getChromeDriverPath();
+	}
+
+	private boolean hasValidRecordingEnvironment() throws IOException {
+		return assertProperty(AgentConfigProvider.TOAST_CHROMEDRIVER_PATH) &&
+				assertProperty(AgentConfigProvider.TOAST_TEST_WEB_INIT_RECORDING_URL) &&
+				assertProperty(AgentConfigProvider.TOAST_TEST_WEB_APP_URL);
+	}
+
+	private void stopListener(ActionEvent e){
+		if(listenerStarted) {
+			browserManager.closeBrowser();
 		}
-		if(nomUrlATester == webAppName || nomUrlATester == recorderName) {
-			return ConfigPanel.testWebAppURL(input, true);
+		else{
+			NotificationManager.showMessage(MainAppMessages.RECORDER_NOT_STARTED).showNotification();
 		}
-		else {
-			if(nomUrlATester ==  chromeDriverName) {
-				return ConfigPanel.testWebAppDirectory(input, true);
-			}
-			else {
-				return false;
-			}	
+	}
+	
+	private void settingsListener(ActionEvent e){
+		ConfigPanel p = configPanelProvider.get();
+		if (p == null) {
+			NotificationManager.showMessage(CommonMessages.PROPERTIES_NOT_DISPLAYED);
 		}
 	}
+	
+	public AgentConfig getConfig() {
+		return webConfigProvider.get();
+	}
+
+	private boolean assertProperty(String property) throws IOException{
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)